src/flb_input_chunk.c

```diff
@@ -1442,6 +1461,14 @@ int flb_input_chunk_append_raw(struct flb_input_instance *in,
         }
     }
 
+    real_size = flb_input_chunk_get_real_size(ic);
+    real_diff = llabs(real_size - pre_real_size);
```

Comment:
Let's stop and think a few minutes about the possible scenarios to determine if `llabs` is the right choice here, I'll enumerate the basic ones but feel free to correct me if I'm wrong or expand if there's one I missed : 

.1 A new chunk is created, `pre_real_size` is zero and `real_size` is whatever the file size is. In this case the result is expected to be a positive number.
.2 Data is appended to a chunk bringing it over FLB_INPUT_CHUNK_FS_MAX_SIZE. Would it be result for the value to be negative in this case? Would using `llabs` means we'd be leaking bytes in the end?

We could probably test this in a controlled environment using the tail input plugin.

Reply:
My personal opinion is that a negative difference is valid and could happen if somehow input_chunk_get triggers some kind of squash or something in chunkio. If that is the case and the way going forward real_diff should be declared as an ssize_t and we would remove the call to llabs.